apps/total-typescript/src/components/exercise-overlay.tsx

```diff
@@ -477,25 +583,61 @@ const handleContinue = async (
   nextExercise: SanityDocument,
   handlePlay: () => void,
   path: string,
+  section?: SanityDocument,
 ) => {
   if (nextExercise._type === 'solution') {
-    const exercise = module.exercises.find((exercise: SanityDocument) => {
-      const solution = exercise.solution
-      return solution?._key === nextExercise._key
-    })
+    if (section) {
```

Comment:
most of the changes are based on checking whether we're inside a section and acting based off of that